authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/web/services/ComponentStatusService.kt

```diff
@@ -23,19 +23,26 @@ import de.sovity.authorityportal.web.utils.TimeUtils
 import io.quarkus.logging.Log
 import io.quarkus.scheduler.Scheduled
 import jakarta.enterprise.context.ApplicationScoped
+import org.eclipse.microprofile.config.inject.ConfigProperty
 import org.jooq.DSLContext
 import org.jooq.impl.DSL
 import java.time.OffsetDateTime
+import java.util.Optional
 
 @ApplicationScoped
 class ComponentStatusService(
     val dsl: DSLContext,
     val uptimeKumaClient: UptimeKumaClient,
-    val timeUtils: TimeUtils
+    val timeUtils: TimeUtils,
+    @ConfigProperty(name = "authority-portal.kuma.api-key") val uptimeKumaApiKey: Optional<String>
 ) {
 
     @Scheduled(every = "30s")
     fun fetchComponentStatuses() {
+        if (uptimeKumaApiKey.isEmpty) {
+            return
+        }
+
```

Comment:
@kamilczaja I see no point scheduling the method if the configProperty is null

Reply:
Unfortunately, Quarkus doesn't have anything like Spring's `ConditionalOnProperty` because something something philosophy.

The closest thing I found is setting up a `SkipPredicate` bean that would be evaluated to determine whether to run the scheduled function or not. But according to documentation, it would result in the same code as what I already wrote anyway.

See: https://quarkus.io/guides/scheduler-reference#conditional_execution